week17/friday/AngularJS-intro.md

```diff
@@ -0,0 +1,191 @@
+
+
+##New Markup
+
+Part of how AngularJS streamlines the logic for dynamic, interactive sites is by 
+adding new markup to the view. There are two main kinds of new markup: curly braces 
+and directives.
+
+###Curly braces
+
+Curly braces are pretty simple: they return the result of the code inside the curly 
+braces. Rails uses some very similar logic in .erb files: `<%= foo %>`. Let’s take a 
+quick glance at what that looks like.
+
+If I put `2 + 2` in my html, all I see on the webpage is essentially a string, 
+‘2 + 2’. The code here isn’t evaluated, it’s just shown. However, if I put it 
+in the curly braces in an AngularJS app, so I type `{{2 + 2}}`, the app will 
+actually evaluate the code inside the curly braces and display the result--so 
+here, I would see "4" instead of "2 + 2" You can use variables, custom functions, 
+and so on here, too.
+
+You'll see and use this more in the activity later today.
+
+###Directives
+
+Directives are a particular AngularJS kind of attribute in your otherwise 
+regular HTML. Just like you can add an href to an anchor tag which your HTML 
+then turns into a link, which looks like `<a href="foo">`, you can add 
+ng- directives to your Angular app which will then run the appropriate logic. 
+You add these to your HTML just like you add regular HTML attributes--you take a 
+view element, like a `<div>`, and add the attribute name like `ng-foo`, an equals 
+sign, and then, in quotation marks, what you want that attribute to equate to. 
+So my div might look like `<div ng-foo="bar">`.
+
+
+There are a heck of a lot of different directives straight out of the box, and 
+we’ll go over several of them over the next week and a half, but right now we’ll
+take a brief look at two of them: `ng-model` and `ng-click`.
+
+####ng-model
+
+Let's look at the following code:
+`<input ng-model="searchTerm"/> {{searchTerm}}`
+
+`ng-model` assigns the value of the element to a variable. In this case, the value
+is what I put in my `input` code: it’s the value the user is inputting. The variable 
+that this value is being assigned to is `searchTerm`. `searchTerm` is being displayed
+to the user in the curly braces. When the user puts something in the input--say, they
+type "help"--the word "help" will immediately be displayed as they type it to the right
+of the input. If `searchTerm` already exists, this new value will replace the old value. 
+If it doesn’t already exist, a new variable will be created.
+
+####ng-click
+
+Let's look at the following code:
+`<button ng-click="alert('Success!')">Did it work?</button>`
+
+`ng-click` runs the code to the right of the equal sign when that HTML element is
+clicked. Sometimes, this is evaluated entirely in the view, like in this instance;
+often, it is a custom method that you define in your controller. In this case, when
+you click on the button, you see an alert.
+
+####Custom directives
+
+In addition to the standard directives that come with AngularJS, you also have the 
```

Comment:
I talk about that some earlier in this section, at around line 138. Is that what you mean?

"Just like you can add an href to an anchor tag which your HTML 
then turns into a link, which looks like `<a href="foo">`, you can add 
ng- directives to your Angular app which will then run the appropriate logic. 
You add these to your HTML just like you add regular HTML attributes" ....etc